File: app/src/main/java/com/example/pjatkpamo/RecipeFragment.java
package com.example.pjatkpamo;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentResultListener;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

public class RecipeFragment extends Fragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_recipe, container, false);

        getParentFragmentManager().setFragmentResultListener("requestKey", this, new FragmentResultListener() {
            @Override
            public void onFragmentResult(@NonNull String requestKey, @NonNull Bundle bundle) {
                double result = bundle.getDouble("bundledKey");

                TextView textViewRecipe = view.findViewById(R.id.textViewRecipe);

                if(result >= 18.5f && result < 25.0f)
                {
                    textViewRecipe.setText(R.string.recpie1);
                }

            }
        });

        return view;
    }
}

File: app/src/main/java/com/example/pjatkpamo/RecipeFragment.kt
package com.example.pjatkpamo

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentResultListener


class RecipeFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {

        val view: View = inflater.inflate(R.layout.fragment_recipe, container, false)

        parentFragmentManager.setFragmentResultListener("requestKey", this,
            FragmentResultListener { requestKey, bundle ->
                val result = bundle.getDouble("bundledKey")
                val textViewRecipe = view.findViewById<TextView>(R.id.textViewRecipe)

                if (result < 25.0f) {
                    textViewRecipe.text = getString(R.string.recipe_high_calories)
                    return@FragmentResultListener
                }

                textViewRecipe.text = getString(R.string.recipe_low_calories)
            })

        return view
    }

}